### DISCUSSION TITLE
Cloudflare Snippets

### DISCUSSION THREAD

**Discussion by Stunt0265** [umami-software/umami#3077] [Q&A]
Is it possible to use Snippets to place the Umami tracking code here vs on the website itself?

https://developers.cloudflare.com/rules/snippets/

**Comment by TheOnlyWayUp** [comment]
Yep, I do this with my reverse proxies instead of with cloudflare (paid feature).

As for snippets, take a look at this [example](https://developers.cloudflare.com/rules/snippets/examples/rewrite-site-links/) from the docs. The is to replace `</head>` with `<script ...></script></head>`.

 I'm not sure what the code would look like exactly, but you should be able to leverage `HTMLRewriter()` to target `</head>` instead. You'd have to rewrite the AttributeRewriter as well.
